<ui version="4.0" >
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>730</width>
    <height>646</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>63</y>
     <width>730</width>
     <height>560</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3" >
    <property name="spacing" >
     <number>0</number>
    </property>
    <property name="margin" >
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter_3" >
      <property name="orientation" >
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QWidget" name="layoutWidget" >
       <layout class="QVBoxLayout" name="verticalLayout_2" >
        <item>
         <widget class="HgRepoView" name="tableView_revisions" >
          <property name="frameShape" >
           <enum>QFrame::StyledPanel</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6" >
          <item>
           <widget class="QFrame" name="frame_branch" >
            <property name="frameShape" >
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow" >
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2" >
             <property name="margin" >
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_4" >
               <property name="text" >
                <string>Branch</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="branch_comboBox" />
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_revrange" >
            <property name="frameShape" >
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow" >
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4" >
             <property name="margin" >
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_3" >
               <property name="enabled" >
                <bool>true</bool>
               </property>
               <property name="text" >
                <string>Revision range</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinbutton_rev_low" />
             </item>
             <item>
              <widget class="QSpinBox" name="spinbutton_rev_high" />
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_filter" >
            <property name="frameShape" >
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow" >
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_5" >
             <property name="margin" >
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2" >
               <property name="text" >
                <string>Filter (regex)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="entry_file_filter" />
             </item>
             <item>
              <widget class="QPushButton" name="button_filter" >
               <property name="text" >
                <string>Filter</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_maincontent" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Expanding" hsizetype="Preferred" >
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape" >
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow" >
        <enum>QFrame::Plain</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitter_2" >
          <property name="orientation" >
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QSplitter" name="splitter" >
           <property name="orientation" >
            <enum>Qt::Vertical</enum>
           </property>
           <widget class="RevDisplay" name="textview_header" />
           <widget class="QFrame" name="textview_frame" >
            <property name="frameShape" >
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow" >
             <enum>QFrame::Raised</enum>
            </property>
           </widget>
          </widget>
          <widget class="QTableView" name="tableView_filelist" />
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_find" >
      <property name="frameShape" >
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow" >
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout" >
       <property name="margin" >
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="close_find_toolButton" >
         <property name="text" >
          <string/>
         </property>
         <property name="icon" >
          <iconset resource="hgqv.qrc" >
           <normaloff>:/icons/close.png</normaloff>:/icons/close.png</iconset>
         </property>
         <property name="flat" >
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label" >
         <property name="text" >
          <string>Find</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="entry_find" />
       </item>
       <item>
        <widget class="QToolButton" name="button_find" >
         <property name="text" >
          <string>Find next</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="button_cancelsearch" >
         <property name="text" >
          <string>Cancel</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer" >
         <property name="orientation" >
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>730</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile" >
    <property name="title" >
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen_repository" />
    <addaction name="actionRefresh" />
    <addaction name="separator" />
    <addaction name="actionQuit" />
   </widget>
   <widget class="QMenu" name="menuHelp" >
    <property name="title" >
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout" />
   </widget>
   <widget class="QMenu" name="menu_Edit" >
    <property name="title" >
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionFind" />
   </widget>
   <addaction name="menuFile" />
   <addaction name="menu_Edit" />
   <addaction name="menuHelp" />
  </widget>
  <widget class="QStatusBar" name="statusbar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>623</y>
     <width>730</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar_edit" >
   <property name="enabled" >
    <bool>true</bool>
   </property>
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>30</y>
     <width>69</width>
     <height>33</height>
    </rect>
   </property>
   <property name="windowTitle" >
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionFind" />
   <addaction name="actionCloseFind" />
  </widget>
  <widget class="QToolBar" name="toolBar_treefilters" >
   <property name="enabled" >
    <bool>true</bool>
   </property>
   <property name="geometry" >
    <rect>
     <x>69</x>
     <y>30</y>
     <width>661</width>
     <height>33</height>
    </rect>
   </property>
   <property name="windowTitle" >
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpen_repository" >
   <property name="text" >
    <string>&amp;Open repository</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionRefresh" >
   <property name="text" >
    <string>&amp;Refresh</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionQuit" >
   <property name="text" >
    <string>&amp;Quit</string>
   </property>
   <property name="toolTip" >
    <string>Quit</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout" >
   <property name="text" >
    <string>About</string>
   </property>
  </action>
  <action name="actionFind" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="text" >
    <string>&amp;Find...</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionCloseFind" >
   <property name="icon" >
    <iconset resource="hgqv.qrc" >
     <normaloff>:/icons/close.png</normaloff>:/icons/close.png</iconset>
   </property>
   <property name="text" >
    <string>close</string>
   </property>
   <property name="visible" >
    <bool>false</bool>
   </property>
  </action>
  <action name="actionFindNext" >
   <property name="text" >
    <string>Find &amp;Next</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HgRepoView</class>
   <extends>QTableView</extends>
   <header>hgrepoview.h</header>
  </customwidget>
  <customwidget>
   <class>RevDisplay</class>
   <extends>QTextBrowser</extends>
   <header>hgrepoview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="hgqv.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>actionFind</sender>
   <signal>toggled(bool)</signal>
   <receiver>frame_find</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>364</x>
     <y>576</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCloseFind</sender>
   <signal>activated()</signal>
   <receiver>actionFind</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFind</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionCloseFind</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
